HW06.Rmd

```diff
@@ -1,92 +1,136 @@
-
-#print out the t-test results
-```
-
-
-
-Now create a graphing function to create boxplots to visualize the results. Plot expression on the y-axis. The graph should look like [this example](IRX1_example.png)
-
-```{r create a graphing function}
-#to work in the tidyverse, it will be easier to make tidy the dataframe first
-
-
-
-#create a graphing function
-
-#then use a for loop combined with the graphing function to make a graph for all your genes of interest 
-```
-
-
-```{r reproducibility}
-sessionInfo()
-```
-
+---
+title: "HW06"
+output: github_document
+---
+
+In cancer, mutations arise that promote growth or survival of cells. In glioblastoma multiforme and other high grade gliomas, a common mutation is a mutation of the 27th lysine (K) to a methionine (M) of the histone subunit H3, or in short H3K27M. 
+
+H3K27M is the most frequent oncohistone in brain cancers, but the biology is still not well understood. Your analysis is to look at the expression of several (27) genes to see if they are differentially expressed and plot 27 boxplots each gene. The data used in this analysis was obtained [from this publication](https://pubmed.ncbi.nlm.nih.gov/31085178/) 
+
+Steps: 
+
+  1. Read in the 45 processed RNA-Seq data found in "./RNA_Seq_processed"
+  2. Map gene.ids to gene.symbols (which I have for you)
+  3. For 27 genes of interest AND your favorite gene, perform a t-test to see if it is differentially expressed between the WT vs H3K27M samples
+  4. Create a graphing function and then create a boxplot that graphs expression between the two groups
+  
+  
+#### Code
+
+From the RNA-Seq files, you only need the "Name" from one file and the "TPM" column from all the files. TPM stands for "transcripts per million" and is a common unit for normalized expression data. 
+```{r load in data}
+#hint, using apply (specifically sapply) you can read in the data into a list and then bind the columns together. Or you can use a for loop too. 
+library(tidyverse)
+h3k27m_list <- list.files("RNA_Seq_processed/H3K27M")
+wildtype_list <- list.files("RNA_Seq_processed/WT")
+
+#create a dataframe with the first set of data from H3K27M files
+full_list <- read_tsv(paste("RNA_Seq_processed/H3K27M/", h3k27m_list[1],sep = ""), col_types = cols()) %>%
+  dplyr::select("Name","TPM")
+colnames(full_list) <- c("gene_id","H3K27M.1")
+
+#add other datasets as columns after the first one
+for (i in 2:length(h3k27m_list)){
+  full_list <- full_list%>%
+    cbind(read_tsv(paste("RNA_Seq_processed/H3K27M/", h3k27m_list[i], sep = ""),col_types = cols()) %>%
+            dplyr::select("TPM"))
+    colnames(full_list)[i+1] <- paste("H3K27M.", i, sep = "")
+}
+
+#get the value of current full list
+j <- length(full_list)
+
+#add wildtype data after the mutant ones
+for (i in 1:length(wildtype_list)){
```

Comment:
This is a very good example of using for loop to read in the files within the same directory!